app/models/market/initial_costs.rb

```diff
@@ -0,0 +1,53 @@
+module Market
+  class InitialCosts
+    REQUIRED = %i(investment_cost technical_lifetime stakeholder)
+
+    def initialize(network)
+      @network = network
+    end
+
+    def calculate
+      technology_costs.merge(topology_costs) do |_, tech_cost, topology_cost|
+        tech_cost + topology_cost
+      end
+    end
+
+    private
+
+    def topology_costs
+      group_sum(topology_nodes) do |node|
+        node.get(:investment_cost) / node.get(:technical_lifetime)
+      end
+    end
+
+    def technology_costs
+      group_sum(technology_nodes) do |node|
+        node.techs.map(&:installed).sum do |tech|
+          tech.initial_investment / tech.technical_lifetime
+        end
+      end
+    end
+
+    def group_sum(nodes)
+      Hash[nodes.group_by{|node| node.get(:stakeholder) }.map do |_, nodes|
```

Comment:
FYI: Naming a variable `_` is [meant to indicate that the variable is not used](https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars). You can also give it a name, but prefixed with `_` [such as here](https://github.com/quintel/etmoses/blob/ee3522363053313835eb391acc095218412d6298/app/models/network/technologies/generic.rb#L60). In this case, the variable *is* used (as the first element of the array on line 33), so it should have a name.

See [Ruby feature #6693](https://bugs.ruby-lang.org/issues/6693).